### ISSUE TITLE
CSI Volume test failing due on newer 20.04 Ubuntu image

### ISSUE THREAD

**Issue by AnishShah** [kubernetes/kubernetes#114221]
### What happened?

We have a Kubernetes cluster on Azure and AWS. Azure cluster is running on Ubuntu ```publisher: "Canonical"
offer: "0001-com-ubuntu-server-focal"
sku: "20_04-lts-gen2"
version: "20.04.202211151"```
 and AWS cluster is running on Ubuntu `ubuntu-minimal/images/hvm-ssd/ubuntu-focal-20.04-amd64-minimal-20221117`
 
Azure cluster is running azuredisk-csi-driver and AWS cluster is running aws-ebs-csi-driver. The following test is failing for both the cluster consistently - 

```
[Testpattern: Dynamic PV (default fs)(allowExpansion)] volume-expand Verify if offline PVC expansion works
```

and we notice the following error in Events - 

```
FailedMount: MountVolume.MountDevice failed while expanding volume for volume "pvc-6e19b0fb-7491-4ec9-bc18-5308c2d8cd8a" : Expander.NodeExpand failed to expand the volume : rpc error: code = Internal desc = Could not resize volume "vol-0ae95d7063d0afae1" ("/dev/sdc"):  resize of device /dev/sdc failed: exit status 1. resize2fs output: resize2fs 1.46.2 (28-Feb-2021)
resize2fs: Superblock checksum does not match superblock while trying to open /dev/sdc
Couldn't find valid filesystem superblock.
```

We created another with an older Ubuntu image (no other changes). Azure cluster is running ```publisher: "Canonical"
offer: "0001-com-ubuntu-server-focal"
sku: "20_04-lts-gen2"
version: "20.04.202210180"``` and AWS cluster is running ```ubuntu-minimal/images/hvm-ssd/ubuntu-focal-20.04-amd64-minimal-20221018```. The same test passed on this cluster.

This could be a Linux kernel issue?

### What did you expect to happen?

I expected the test to pass on the newer 20.04 Ubuntu image.

### How can we reproduce it (as minimally and precisely as possible)?

Create a cluster with the above Ubuntu image and run CSI tests against AWS EBS and AzureDisk CSI driver test.

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

1.25.4

### Cloud provider

<details>

</details>
 AWS and Azure

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

AWS: 5.15.0-1023-aws
Azure: 5.15.0-1023-azure

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by andyzhangx** [comment]
@AnishShah thanks for reporting. On Azure Disk CSI driver, we have worked out this PR to workaround the second resize failure issue: https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1728

**Comment by AnishShah** [comment]
We were able to reproduce this issue without any Kubernetes layer which concluded that this was an issue in the Linux kernel. We are working with Canonical and they plan to fix this in upcoming Ubuntu images.

**Comment by andyzhangx** [comment]
> We were able to reproduce this issue without any Kubernetes layer which concluded that this was an issue in the Linux kernel. We are working with Canonical and they plan to fix this in upcoming Ubuntu images.

@AnishShah that would require quite long time for backporting the fix, the above PR could fix the issue even without kernel fix.

**Comment by AnishShah** [comment]
Yes @andyzhangx, correct. I agree. I was just providing an update about the underlying issues.

**Comment by dennis-menge** [comment]
> We were able to reproduce this issue without any Kubernetes layer which concluded that this was an issue in the Linux kernel. We are working with Canonical and they plan to fix this in upcoming Ubuntu images.

Can you share the work item on the Ubuntu side?

**Comment by AnishShah** [comment]
The issue is fixed in `linux-aws 5.15.0.1031.35~20.04.20` AMI. Similar images should be available in Azure and GCP
